src/coins.cpp

```diff
@@ -260,6 +295,196 @@ void CCoinsViewCache::BringBestAnchorIntoCache(
     assert(GetSaplingAnchorAt(currentRoot, tree));
 }
 
+void draftMMRNode(std::vector<uint32_t> &indices, 
+                  std::vector<HistoryEntry> &entries, 
+                  HistoryNode nodeData, 
+                  uint32_t h, 
+                  uint32_t peak_pos) 
+{
+    HistoryEntry newEntry = h == 0 
+        ? libzcash::NodeToEntry(nodeData) 
+        : libzcash::NewEntry(nodeData, peak_pos - (1 << h) - 1, peak_pos - 2);
```

Comment:
Explain in a comment what `peak_pos - (1 << h) - 1` and `peak_pos - 2` are.

Reply:
It is instantiating new entry that has childs and their indices

 `peak_pos - (1 << h) - 1` is a left child index (starts from 0, not 1, thus -1)
`peak_pos - 2` is the right child index (also starts from 0, thus - 2)

`peak_pos` itself is mmr position (which starts from 1), so that's why those off-by-one corrections

Reply:
Why does the MMR position start from 1? I recommend always starting numbering from 0. https://www.cs.utexas.edu/users/EWD/transcriptions/EWD08xx/EWD831.html

Reply:
@daira  Below are a lot of places which work assuming that peak positions are starting from 1. Most of them are based on zip pseudocode, so I want to keep them close to zip when possible.
Only when it comes to indexing to the array representation, they are converted to zero-based array indices.